awesome/src/objects/tag.rs

```diff
@@ -69,29 +69,23 @@ impl<'lua> Tag<'lua> {
                           .position(|c| *c == *client))
     }
 
-    pub fn tag_client(&mut self, obj: Client<'lua>) -> rlua::Result<Value> {
-        let client = Client::cast(obj.clone().into())?;
+    pub fn tag_client(&mut self, client: Client<'lua>) -> rlua::Result<()> {
         if let Some(_) = self.client_index(&client)? { // if it is already part of the clients
-            return Ok(Value::Nil);
+            return Ok(());
         }
         let mut clients = self.clients()?;
-        clients.push(obj);
+        clients.push(client);
         self.set_associated_data("__clients", clients)?;
-        Ok(Value::Nil)
+        Ok(())
     }
 
-    pub fn untag_client(&mut self, obj: Client<'lua>) -> rlua::Result<Value> {
-        let client = Client::cast(obj.clone().into())?;
-        let mut clients = self.clients()?;
-
-        match self.client_index(&client)? {
-            Some(index) => {
-                clients.remove(index);
-                self.set_associated_data("__clients", clients)?;
-                Ok(Value::Nil)
-            }
-            None => Err(rlua::Error::RuntimeError("Client to untag was not tagged".into()))
-        }
+    pub fn untag_client(&mut self, client: Client<'lua>) -> rlua::Result<()> {
+        let clients: Vec<_> = self.clients()?
+                              .into_iter()
+                              .filter(|c| *c != client)
```

Comment:
This is not the same logic as there should be an error signaled if the client wasn't in the list

Reply:
In fact it is closer to Awesome's bahaviour

```
void
untag_client(client_t *c, tag_t *t)
{
    for(int i = 0; i < t->clients.len; i++)
        if(t->clients.tab[i] == c)
        {
            lua_State *L = globalconf_get_lua_State();
            client_array_take(&t->clients, i);
            banning_need_update();
            ewmh_client_update_desktop(c);
            screen_update_workarea(c->screen);
            tag_client_emit_signal(t, c, "untagged");
            luaA_object_unref(L, t);
            return;
        }
}
```